pytensor/tensor/subtensor.py

```diff
@@ -489,8 +489,10 @@ def indexed_result_shape(array_shape, indices, indices_are_shapes=False):
     remaining_dims = range(pytensor.tensor.basic.get_vector_length(array_shape))
     idx_groups = group_indices(indices)
 
-    if len(idx_groups) > 2 or len(idx_groups) > 1 and not idx_groups[0][0]:
-        # Bring adv. index groups to the front and merge each group
+    if len(idx_groups) > 3 or len(idx_groups) == 3 and not idx_groups[0][0]:
```

Comment:
I had to look up operator precedence for `and` and `or` here. :-)
Turns out and has higher precedence, which I think is correct here.
Is that just me, or should we maybe add parenthesis here to make it obvious?

Reply:
Always in favor of parenthesis. Brings us closer to lisp!